plugins/browser-plugin-webview/src/index.ts

```diff
@@ -0,0 +1,70 @@
+import { BrowserPlugin, BrowserTracker, Payload } from '@snowplow/browser-tracker-core';
+import { hasMobileInterface, trackWebViewEvent } from '@snowplow/webview-tracker';
+import { Logger, SelfDescribingEvent, SelfDescribingJson } from '@snowplow/tracker-core';
+
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function WebViewPlugin(): BrowserPlugin {
+  let LOG: Logger;
+
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+
+    logger: (logger: Logger) => {
+      LOG = logger;
+    },
+
+    filter: (payload) => {
+      if (hasMobileInterface() === true) {
+        LOG.debug(`Payload (event ID: ${payload.eid}) was filtered out and forwarded to WebView tracker.`);
+
+        let atomicProperties = {
+          eventName: payload.e as string,
+          trackerVersion: payload.tv as string,
+          useragent: payload.ua as string,
+          url: payload.url as string,
+          title: payload.page as string,
+          referrer: payload.refr as string,
+          category: payload.se_ca as string,
+          action: payload.se_ac as string,
+          label: payload.se_la as string,
+          property: payload.se_pr as string,
+          value: payload.se_va as number,
+          minXOffset: payload.pp_mix as number,
+          maxXOffset: payload.pp_max as number,
+          minYOffset: payload.pp_miy as number,
+          maxYOffset: payload.pp_may as number,
+        };
+        let event = getSelfDescribingEventData(payload);
+        let entities = getEntities(payload);
+        let trackers: Array<string> = Object.keys(_trackers);
+
+        trackWebViewEvent({ properties: atomicProperties, event: event, context: entities }, trackers);
+
+        return false;
+      } else {
+        return true;
+      }
+    },
+  };
+}
+
+function getSelfDescribingEventData(payload: Payload): SelfDescribingEvent | undefined {
+  if (payload.ue_pr) {
+    return JSON.parse(payload.ue_pr as string);
+  } else if (payload.ue_px) {
+    return JSON.parse(payload.ue_px as string);
```

Comment:
You can't just parse a base64 encoded string to JSON directly, you will first need to decode it. Unfortunately the function is not exported but there is a `base64urldecode` function in the core library that you could copy ([see here](https://github.com/snowplow/snowplow-javascript-tracker/blob/743e77df8bf914682017e350469b99c22207505f/libraries/tracker-core/src/payload.ts#L189))


The same in `getEntities`. I think this is worth a unit test.